### ISSUE TITLE
unusedArguments rule doesn't handle new shorthand optional binding syntax properly

### ISSUE THREAD

**Issue by calda** [nicklockwood/SwiftFormat#1166]
The `unusedArgument` rule doesn't handle the new shorthand optional binding syntax properly. For example it incorrectly updates this sample (which [compiles in Swift 5.7](https://swiftfiddle.com/?c=H4sIAAAAAAAAA6tWSlayUkorzUtWyE0tychP0YhXSMvPt1LwzCux11SojslTUMhMU8hJLQEJK1Qr1Mbk1SrpKJUBdeVlpmeU5FTq5iZm5inVAgDFHOm8SgAAAA%3D%3D)) to no longer compile:

### Before

```swift
func method(_ foo: Int?) {
  if let foo {}
}
```

### After

```swift
func method(_: Int?) {
    if let foo {}
}
```

You can find an non-trivial example of this behaving incorrectly by running the new rule from #1165 on the regression suite:

```diff
@@ -74,8 +74,8 @@ public enum LayoutError: Error, Hashable, CustomStringConvertible {
     case unknownSymbol(Error /* SymbolError */, [String])
     case multipleMatches([URL], for: String)
 
-    public init?(_ error: Error?) {
-        guard let error = error else {
+    public init?(_: Error?) {
+        guard let error else {
             return nil
         }
         self.init(error)
```

**Comment by nicklockwood** [comment]
@calda fixed in 0.49.7

**Comment by calda** [comment]
Speedy!!!
